src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -651,68 +651,135 @@ private static object SplitOperatorImpl(ExecutionContext context, IScriptExtent
+                                if ((strIndex + 1) < item.Length)
+                                {
+                                    split.Add(item.Substring(strIndex + 1));
+                                }
+                                else
+                                {
+                                    split.Add(string.Empty);
+                                }
+
+                                break;
                             }
-                            else
+
+                            // If this delimiter is at the end of the string,
+                            // add an empty string to denote the item "after"
+                            // it.
+                            if (strIndex == (item.Length - 1))
                             {
                                 split.Add(string.Empty);
                             }
-
-                            break;
                         }
-
-                        // If this delimiter is at the end of the string,
-                        // add an empty string to denote the item "after"
-                        // it.
-                        if (strIndex == (item.Length - 1))
+                        else
                         {
-                            split.Add(string.Empty);
+                            buf.Append(item[strIndex]);
                         }
                     }
-                    else
+                }
+                else 
+                {
+                    for (int strIndex = item.Length - 1; strIndex >= 0; strIndex--)
                     {
-                        buf.Append(item[strIndex]);
+                        object predicateResult = predicate.DoInvokeReturnAsIs(
+                            useLocalScope: true,
+                            errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToExternalErrorPipe,
+                            dollarUnder: CharToString(item[strIndex]),
+                            input: AutomationNull.Value,
+                            scriptThis: AutomationNull.Value,
+                            args: new object[] { item, strIndex });
+                        if (LanguagePrimitives.IsTrue(predicateResult))
+                        {
+                            if (buf.Length != 0) 
+                            {
+                                split.Add(ReverseStringUtility(buf.ToString()));
+                            }
+
+                            buf.Clear();
+
+                            if ((limit * -1) > 0 && split.Count >= ((limit * -1) - 1))
+                            {
+                                // We're one item below the limit. If
+                                // we have any string left, go ahead
+                                // and add it as the last item, otherwise
+                                // add an empty string if there was
+                                // a delimiter at the end.
```

Comment:
You can probably squash this block of comments into two or three lines at most. 🙂